homeassistant/components/api/__init__.py

```diff
@@ -118,8 +118,14 @@ def get(self, request: web.Request) -> web.Response:
         Home Assistant core is running. Its primary use case is for supervisor
         to check if Home Assistant is running.
         """
-        hass = request.app[KEY_HASS]
-        return self.json({"state": hass.state.value})
+        hass: HomeAssistant = request.app[KEY_HASS]
+        migration = recorder.async_migration_in_progress(hass)
+        live_migration = recorder.async_migration_is_live(hass)
+        if migration and not live_migration:
+            state = "MIGRATING"
+        else:
+            state = hass.state.value
+        return self.json({"state": state})
```

Comment:
I don't think it is a good idea to extend the state just for this API call. This can be rather confusing for API users. Either we make it a true Core state, otherwise I'd just extend the dictionary:

```suggestion
        return self.json({"state": state, "is_migrating": migration and not live_migration})
```

or just if it is migrating in general


```suggestion
        return self.json({"state": state, "is_migrating": migration})
```

Since we anyways need to change Supervisor side, we can simply go with something like

```python
state = data.get("state", "RUNNING")
is_migrating = data.get("is_migrating", False)

return state == "RUNNING" or is_migrating
```